.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,74 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
I think we wanna run that check on every push, don't we?

Reply:
Well we do run it on pull_request, ain't that good enough? You can [see](https://github.com/canonical/charmed-5g-nrf-operator/actions/runs/4733425656/jobs/8400941502?pr=1) the result of the analysis on this PR for example

Reply:
Sure, my point is that if there are some problems, we could potentially detect them faster by running this check on every push.

Reply:
In all the projects we do it like it is here. If we want to change that we can discuss it but it wouldn't be specific to this project. 

How much faster would it be? I personally haven't felt that running codeql on pull_request made me go slower.